File: worker_wakeup.pl
#!/usr/bin/perl

use JSON;

# /* your setting start */

my $app = q{YOUR APP NAME};

# get token command is $ heroku auth:token
my $token = q{YOUR API TOKEN};

my $dyno_name = q{worker}; # worker or bot

my $curl = q{/usr/local/bin/curl};

# /* your setting end */

my $dynos_url = $curl . q{ -s -X GET https://api.heroku.com/apps/} . $app . qq{/dynos -H "Accept: application/vnd.heroku+json; version=3" -H "Authorization: Bearer } . $token . q{"};


my $res = `$dynos_url`;

my $json = JSON::from_json($res);

my $status = $json->[0];

my $state = $status->{'state'};

if ($state && $state eq 'idle') {
	my $restart_url = $curl . q{ -s -X DELETE https://api.heroku.com/apps/} . $app . q{/dynos/} . $dyno_name . qq{ -H "Accept: application/vnd.heroku+json; version=3" -H "Authorization: Bearer } . $token . q{"};
	my $res = `$restart_url`;
}

exit;

File: worker_wakeup.php
<?php

/* your setting start */

$app = 'YOUR APP NAME';

/* get token command is $ heroku auth:token  */

$token = 'YOUR API TOKEN';

$dyno_name = 'worker'; // worker or bot

/* your setting end */

$dynos_url = 'https://api.heroku.com/apps/' . $app . '/dynos';

$headers = array('Authorization: Bearer ' . $token ,'Accept: application/vnd.heroku+json; version=3');

$ch = curl_init( );
curl_setopt( $ch, CURLOPT_URL, $dynos_url);
curl_setopt( $ch, CURLOPT_HTTPHEADER, $headers);
curl_setopt( $ch, CURLOPT_RETURNTRANSFER, true);
$res = curl_exec($ch); 

$json = json_decode( $res, true);

$state = $json[0]['state'];

if ( $state and $state == 'idle') {
	$restart_url = 'https://api.heroku.com/apps/' . $app . '/dynos/' . $dyno_name;
	$ch = curl_init( );
	curl_setopt( $ch, CURLOPT_URL, $restart_url);
	curl_setopt( $ch, CURLOPT_HTTPHEADER, $headers);
	curl_setopt( $ch, CURLOPT_CUSTOMREQUEST, 'DELETE');
	curl_setopt( $ch, CURLOPT_RETURNTRANSFER, true);
	$res = curl_exec($ch); 
}

?>